src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+package parser;
+
+import tasklist.*;
+import exception.*;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * The Parser class is responsible for handling input commands.
+ */
+public class Parser {
+
+    /**
+     * Handles commands regarding tasks(ie. todos, deadlines, events).
+     * @param input The command input.
+     * @param newList The current TaskList.
+     * @throws DukeException if there is an error with the input or input is not given in the correct format.
+     */
+    public static String commandTask(String input, TaskList newList) throws DukeException {
```

Comment:
This method goes beyond 30 LOC (lines of code). Maybe parts of the method could be abstracted out?